packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -265,6 +266,13 @@ class Config:
     provider is selected.
     """
 
+    server_settings: ClassVar[Optional[MappingProxyType[str, Any]]] = None
```

Comment:
Can you check if there's a way to raise an exception when this class attr more than once? This is not a PR blocker, but it would be a good sanity check to include for developers that are accidentally assigning to this class attr again in their own server extension.

Reply:
I agree that this would be nice to have. It would be trivial for an instance attribute with a property descriptor. I think it is feasible for a class instance too by using a metaclass. Taking a look

Reply:
I will merge this PR and open a separate follow-up PR to improve this.